### PR TITLE
Implement the plugin commands

### PR DESCRIPTION
**Pull request by alessio-perugini**
SSIA

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -1,36 +1,258 @@
...
+	}
+	defer rebootFile.Remove()
+
+	slog.Info("sending serial reset")
+	if err := serialutils.TouchSerialPortAt1200bps(portAddress); err != nil {
+		return err
+	}
+
+	slog.Info("uploading command sketch with bossac")
+	cmd, err := executils.NewProcess(nil, p.bossacBin.String(), "-i", "-d", "--port="+portAddress, "-U", "true", "-i", "-e", "-w", "-v", rebootFile.String(), "-R")
```
Comment (umbynos) [inline main.go]: Actually not all the boards use bossac, for example the RP2040 use rp2040tools

File: main.go [hunk 2]
```diff
@@ -1,16 +1,61 @@
...
+	date          = ""
+
+	//go:embed sketches/commands/build/arduino.mbed_nano.nanorp2040connect/commands.ino.bin
+	nanorp2040connectCommandSketchBinary embed.FS
+
+	//go:embed sketches/commands/build/arduino.megaavr.uno2018/commands.ino.bin
+	uno2018CommandSketchBinary embed.FS
+
+	//go:embed sketches/commands/build/arduino.samd.mkrvidor4000/commands.ino.bin
+	mkrvidor4000CommandSketchBinary embed.FS
```
Comment (umbynos) [inline main.go]: drop it

File: sketches/commands/compile.sh
Comment (umbynos) [inline sketches/commands/compile.sh:1]: maybe would be better to create a task for this?

File: Taskfile.yml [hunk 3]
```diff
@@ -93,7 +93,7 @@ tasks:
     desc: Build the Go code
     dir: "{{.DEFAULT_GO_MODULE_PATH}}"
     cmds:
-      - go build -v {{.LDFLAGS}}
+      - CGO_ENABLED=0 go build -v {{.LDFLAGS}}
```
Comment (umbynos) [inline Taskfile.yml:96]: Are we sure that the serial operations are working correctly with CGO disabled? macOS I'm looking at you...

File: main.go [hunk 4]
```diff
@@ -1,36 +1,259 @@
...
 	helper "github.com/arduino/fwuploader-plugin-helper"
 	paths "github.com/arduino/go-paths-helper"
+	"github.com/arduino/uno-r4-wifi-fwuploader-plugin/serial"
+	serialutils "github.com/arduino/uno-r4-wifi-fwuploader-plugin/serial/utils"
 	semver "go.bug.st/relaxed-semver"
+	"golang.org/x/exp/slog"
+)
+
+const (
+	pluginName = "nina-fwuploader"
```
Comment (umbynos) [inline main.go:37]:
```suggestion
	pluginName = "nina-fwuploader-plugin"
```

**Review by umbynos — CHANGES_REQUESTED** [review]

File: main.go (hunk 4 above)
Comment (umbynos) [reply main.go:37]: nevermind: https://github.com/arduino/uno-r4-wifi-fwuploader-plugin/blob/af756953af1e9cb74191f1909e6354fd6ca6adcb/main.go#L39

File: Taskfile.yml (hunk 3 above)
Comment (alessio-perugini) [reply Taskfile.yml:96]: Nope, we have to test it on mac. Unfortunately I don't have any at home :cry:

Comment (umbynos) [reply Taskfile.yml:96]: @cmaglie could you please help with this?

**Review by umbynos — APPROVED** [review]
Nice Work!!

File: Taskfile.yml (hunk 3 above)
Comment (cmaglie) [reply Taskfile.yml:96]: Tested on MacOSX Venutra 13.4.1 (Apple M1) and an Arduino Nano 33 IoT:
- ✅ Firmware Upload
- ✅ Firmware get-version
- ✅ Certificate upload

**Review by cmaglie — APPROVED** [review]

**Pull request merged**